README.md

```diff
@@ -355,6 +356,75 @@ writable.
 
 Always prefer `ReadonlyArray` over a regular `Array` unless it must be possible to modify the Array.
 
+### React props & TypeScript
+
+#### Why use Interface instead of Type for typing props?
+
+`interface` are a little more powerful than the corresponding `type` declaration.
+
+One of the reasons to use `interface` is when you need to compose two or more types, you have the
+option of extending those types with an interface, or intersecting them in a type alias. TS team
+says:
+
+> Interfaces create a single flat object type that detects property conflicts, which are usually
+> important to resolve! Intersections on the other hand just recursively merge properties, and in
+> some cases produce never. Interfaces also display consistently better, whereas type aliases to
+> intersections can't be displayed in part of other intersections. Type relationships between
+> interfaces are also cached, as opposed to intersection types as a whole. A final noteworthy
+> difference is that when checking against a target intersection type, every constituent is checked
+> before checking against the "effective"/"flattened" type. For this reason, extending types with
+> interfaces/extends is suggested over creating intersection types.
+
+This is preferred:
+
+```ts
+interface Foo extends Bar, Baz {
+  someProp: string;
+}
+```
+
+Instead of this:
+
+```ts
+type Foo = Bar &
+  Baz & {
+    someProp: string;
+  };
+```
+
+#### Why not use React.FC or FunctionComponent type?
+
+- Provides an implicit definition of `children`
+- Doesn't support generics.
+- Doesn't work correctly with `defaultProps`
```

Comment:
nitpick, both this bulletin and the first one are missing `.`.
I wonder if we even care if it works or doesn't with `defaultProps` as we can define default props in the destruction :arrow_down: 
```ts
interface FancyButtonProps {
  color: string;
}

function FancyButton({ color = "blue" }: FancyButtonProps) => {
  //...
};
```

Reply:
@psimk Hi!

Yeah, but those annotations were added for letting know why not use React.FC or Function Component type. For reinforce the idea of typing the props using the Interface.

source: https://github.com/facebook/create-react-app/pull/8177